src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -502,9 +499,10 @@ SHAMap::onlyBelow (SHAMapTreeNode* node) const
 
     // An inner node must have at least one leaf
     // below it, unless it's the root_
-    assert (node->hasItem () || (node == root_.get ()));
+    auto leaf = static_cast<SHAMapTreeNode*>(node);
+    assert (leaf->hasItem () || (leaf == root_.get ()));
```

Comment:
I'm happy to be wrong, but I just don't see anywhere `root_` is assigned to point to anything but a `SHAMapInnerNode`. What I missed was that the loop can't exit if the node is not a leaf. I also missed that the only _caller_ to `onlyBelow` explicitly checks that it's not looking at the root.

The function is safe (and there's no real risk of infinite loop either, unless there's a loop in the tree or something incomparably bad), but I don't think that the assertion will ever get to this line with `leaf == root_get()`, at least as currently used.